packages/mux-video/src/ads/mixin.ts

```diff
@@ -0,0 +1,426 @@
+        <style>
+          :host {
+            aspect-ratio: var(--media-aspect-ratio, 16 / 9);
+            display: inline-block;
+            line-height: 0;
+            width: 100%;
+            height: 100%;
+            display: block;
+          }
+
+          video {
+            display: block;
+            max-width: 100%;
+            max-height: 100%;
+            min-width: 100%;
+            min-height: 100%;
+            object-fit: var(--media-object-fit, contain);
+            object-position: var(--media-object-position, 50% 50%);
+          }
+
+          video::-webkit-media-text-track-container {
+            transform: var(--media-webkit-text-track-transform);
+            transition: var(--media-webkit-text-track-transition);
+          }
+
+          #mainContainer {
+              position: relative;
+              width: 100%;
+              height: 100%;
+          }
+
+          #adContainer {
+              position: absolute;
+              top: 0px;
+              left: 0px;
+              bottom: 0px;
+              right: 0px;
+              z-index: -1;
+              width: 100%;
+              height: 100%;
+          }
+
+          #mainContainer #adContainer.ad-playing {
+            z-index: 2;
+          }
+
+          #imaUnavailableMessage {
+            position: absolute;
+            top: 0;
+            left: 0;
+            z-index: 10;
+            background: rgba(0, 0, 0, 0.75);
+            color: white;
+            font-size: 0.9em;
+            text-align: center;
+            line-height: 1.4;
+            width: 100%;
+            height: 100%;
+            align-items: center;
+            align-content: center;
+            cursor: not-allowed;
+          }
+
+          #imaUnavailableMessage h4 {
+            font-size: 1rem;
+            margin: 0;
+          }
+        </style>
+        <div id="mainContainer">
+            <slot name="media">
+              <video id="contentElement" ${serializeAttributes(attrs)}></video>
+            </slot>
+          <div id="adContainer"></div>
+        </div>
+        <slot></slot>
+      `;
+    };
+
+    constructor(...args: any[]) {
+      super(...args);
+
+      const resizeObserver = new ResizeObserver((entries) => {
+        for (const entry of entries) {
+          const { width, height } = entry.contentRect;
+          if (width > 0 && height > 0) {
+            this.#adProvider?.updateAdsManagerSize(width, height);
+          }
+        }
+      });
+      resizeObserver.observe(this);
+    }
+
+    connectedCallback(): void {
+      super.connectedCallback();
+
+      if (!GoogleImaAdsProvider.isGoogleImaSDKAvailable()) {
+        console.error('Missing google.ima SDK. Make sure you include it via a script tag.');
+        if (!this.allowAdBlocker) {
+          this.#showAdBlockedMessage();
+        } else {
+          this.#adBreak = false;
+        }
+        return;
+      }
+
+      const config: GoogleImaAdsProviderConfig = {
```

Comment:
thought(non-blocking): we should probably iterate on several of these implementation/naming details but this should be treated as lower priority and definitely as a followup since it's currently just for "under the hood" logic in (extended) mux-video.